src/full/Agda/Syntax/Common.hs

```diff
@@ -2903,65 +2903,98 @@ instance NFData ExpandedEllipsis where
   rnf NoEllipsis             = ()
 
 -- | Notation as provided by the @syntax@ declaration.
-type Notation = [GenPart]
+type Notation = [NotationPart]
```

Comment:
Are there any invariants in `Notation`, like alternation between holes and non-holes or something?  If yes, please add a comment.

Reply:
I don't know exactly what the invariants are.